GoMoney/Extensions/Date+Extension.swift

```diff
@@ -0,0 +1,53 @@
+import Foundation
+
+extension Date {
+    func getLast6Month() -> Date? {
+        return Calendar.current.date(byAdding: .month, value: -6, to: self)
+    }
+    
+    func getLast3Month() -> Date? {
+        return Calendar.current.date(byAdding: .month, value: -3, to: self)
+    }
+    
+    func getYesterday() -> Date? {
+        return Calendar.current.date(byAdding: .day, value: -1, to: self)
+    }
+    
+    func getLast7Day() -> Date? {
+        return Calendar.current.date(byAdding: .day, value: -7, to: self)
+    }
+    
+    func getLast30Day() -> Date? {
+        return Calendar.current.date(byAdding: .day, value: -30, to: self)
+    }
+    
+    func getPreviousMonth() -> Date? {
+        return Calendar.current.date(byAdding: .month, value: -1, to: self)
+    }
+    
+    func getThisMonthStart() -> Date? {
+        let components = Calendar.current.dateComponents([.year, .month], from: self)
+        return Calendar.current.date(from: components)!
```

Comment:
don't force unwrap. return value is Date?, so can return nil
```suggestion
        return Calendar.current.date(from: components)
```